nixos/modules/services/network-filesystems/openafs-server.nix

```diff
@@ -0,0 +1,119 @@
+    printf "${concatStringsSep "\n" (map ({ address, hostname }: concatStringsSep " \#" [ address hostname ]) cfg.cellServers)}\n" >> $out
+  '';
+  bosConfig = pkgs.writeText "BosConfig" ''
+    restarttime 16 0 0 0 0
+    checkbintime 3 0 5 0 0
+    bnode simple vlserver 1
+    parm ${openafsPkgs}/libexec/openafs/vlserver
+    end
+    bnode simple ptserver 1
+    parm ${openafsPkgs}/libexec/openafs/ptserver -db /var/openafs/prdb -restricted
+    end
+    bnode dafs dafs 1
+    parm ${openafsPkgs}/libexec/openafs/dafileserver -vattachpar 128 -vhashsize 11 -cb 1000000
+    parm ${openafsPkgs}/libexec/openafs/davolserver
+    parm ${openafsPkgs}/libexec/openafs/salvageserver
+    parm ${openafsPkgs}/libexec/openafs/dasalvager
+    end
+  '';
+in
+{
+  options = {
+
+    services.openafsServer = {
+
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable the OpenAFS server.";
+      };
+
+      adminUsers = mkOption {
+        type = types.listOf types.string;
+        default = [];
+        description = "List of OpenAFS users that can administer this server.";
+      };
+
+      advertisedAddresses = mkOption {
+        default = [];
+        description = "List of IP addresses this server is advertised under. See NetInfo(5)";
+      };
+      bosConfig = mkOption {
+        type = types.package;
+        default = bosConfig;
+        description = "The server configuration file specifying the started servers and options";
+      };
+
+      cellName = mkOption {
+        default = "grand.central.org";
+        description = "Server's cell name.";
+      };
+
+      cellServers = mkOption {
+        default = [];
+        description = "IP addresses of database servers that serve this cell.";
+        example = literalExample ''
+          [ { address = "10.0.0.3"; hostname = "afsdb.example.org"; } ]
+        '';
+      };
+
+      openafsPkgs = mkOption {
+        default = config.boot.kernelPackages.openafs;
+        description = "The package to use for the openafs servers";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ openafsPkgs ];
+
+    system.activationScripts.openafsServer = stringAfter
+    [ (if config.services.openafsClient.enable then "openafs" else "etc") ] ''
+      mkdir -p /etc/openafs/server
+      cp -r ${afsConfig}/* /etc/openafs/server
+    ''; #*/
+
+    system.activationScripts.openafServerVar = mkIf (netInfo != "") (stringAfter [ "var" ] ''
+      mkdir -p /var/openafs
+      rm -f /var/openafs/NetInfo
+      cp "${netInfo}" /var/openafs/NetInfo
+    '');
+    systemd.services = {
+      openafs-server = {
+        description = "OpenAFS server";
+        after = [ "syslog.target" "network.target" ];
+        wantedBy = [ "multi-user.target" ];
+        unitConfig = {
+          ConditionFileNotEmpty = [ "|/etc/openafs/server/rxkad.keytab"
+                                    "|/etc/openafs/server/KeyFile" ];
```

Comment:
As the names suggest these are key files, which is kerberos terminology for an equivalent of a password stored in a file. You can set up a kerberos server. For a kerberos realm EXAMPLE.COM you create a principal (equiv. to a login) named afs/example.com@EXAMPLE.COM in kerberos with a random password. `kadmin.local addprinc -randkey afs/example.com@EXAMPLE.COM`. And export it with `kadmin.local ktadd -k /etc/openafs/server/rxkad.keytab afs/example.com@EXAMPLE.COM`. I am not sure about the exact syntax but along these lines... The lowercase example.com is the AFS cell. After storing that the afs server should be able to start. (You need a set up and running kerberos server, still)